Expand options documentation test and examples for Ruby

aguspe opened the pull request:
## **User description**
### Description
This PR adds extra tests for the options in ruby and updates the documentation examples, besides that, it also moves the proxy example from the documentation file

### Motivation and Context
For new and experienced user to be able to fully understand all the options offered by Selenium is important to have up-to-date documentation with examples

### Types of changes
- [ ] Change to the site (I have double-checked the Netlify deployment, and my changes look good)
- [x] Code example added (and I also added the example to all translated languages)
- [ ] Improved translation
- [ ] Added new translation (and I also added a notice to each document missing translation)

### Checklist
- [x] I have read the [**contributing**](https://selenium.dev/documentation/en/contributing/) document.
- [x] I have used [hugo](https://gohugo.io/) to render the site/docs locally and I am sure it works.


___

## **Type**
Enhancement, Tests, Documentation


___

## **Description**
- Added multiple new tests in Ruby for Selenium WebDriver options including handling untrusted certificates, unhandled prompts, window rect, strict file interactability, and proxy settings.
- Updated English, Japanese, Brazilian Portuguese, and Chinese WebDriver documentation to link to the new Ruby code examples.



___



## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>options_spec.rb</strong><dd><code>Expand Selenium WebDriver Options Tests in Ruby</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/ruby/spec/drivers/options_spec.rb

<li>Added tests for handling untrusted certificates, unhandled prompts, <br>window rect, strict file interactability, and proxy settings in Chrome <br>and Firefox.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1679/files#diff-14ec885660d66546724e448a9870b3afdd72ec096c7d4777539e3e4dfb65c39c">+45/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>options.en.md</strong><dd><code>Update English WebDriver Options Documentation with Ruby Examples</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/drivers/options.en.md

<li>Updated Ruby code examples with links to the new tests in the <br>options_spec.rb.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1679/files#diff-62bab9f28d27a7f462881a28eca384a0e534a730f849248f396d171820bc748e">+5/-10</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>options.ja.md</strong><dd><code>Update Japanese WebDriver Options Documentation with Ruby Examples</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/drivers/options.ja.md

<li>Updated Japanese WebDriver documentation to include new Ruby test <br>examples.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1679/files#diff-2ed475220bde1368cfa3883d71db179409ddaf9ccbaea847ed3f6ffe816321e4">+5/-11</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>options.pt-br.md</strong><dd><code>Update Brazilian Portuguese WebDriver Options Documentation with Ruby </code><br><code>Examples</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/drivers/options.pt-br.md

<li>Updated Brazilian Portuguese WebDriver documentation to include new <br>Ruby test examples.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1679/files#diff-6eaa598a13e9d64c6afbd54b84aef470520557f08c4d183ffb28ceb9ed04d0bb">+5/-10</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>options.zh-cn.md</strong><dd><code>Update Chinese WebDriver Options Documentation with Ruby Examples</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/drivers/options.zh-cn.md

<li>Updated Chinese WebDriver documentation to include new Ruby test <br>examples.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1679/files#diff-9b23b749355db8154bfcc4a1d70f809b7d4e91a7b3044f9ac74a6c1a005873d6">+5/-11</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

aguspe committed:
Expand options documentation test and examples

aguspe committed:
Merge branch 'trunk' into expand_options_example_page

aguspe commented:
Hello @diemol one question about the documentation, does it make sense that the timeouts are under the options page and not on its page?

If it does, I can also update the timeouts example on this PR

<img width="1192" alt="Screenshot 2024-04-19 at 20 40 42" src="https://github.com/SeleniumHQ/seleniumhq.github.io/assets/33221555/b9754131-2987-4cf2-b371-e45433e48c4f">

aguspe committed:
Merge branch 'trunk' into expand_options_example_page

aguspe committed:
Merge branch 'trunk' into expand_options_example_page

aguspe committed:
Merge branch 'trunk' into expand_options_example_page

aguspe committed:
Merge branch 'trunk' into expand_options_example_page

aguspe committed:
Merge branch 'trunk' into expand_options_example_page

harsha509 commented on examples/ruby/spec/drivers/options_spec.rb:
```diff
@@ -45,5 +45,50 @@
+      options.unhandled_prompt_behavior = :accept
+
+      driver = Selenium::WebDriver.for :chrome, options: options
+      driver.get('https://www.google.com')
```
can we use https://www.selenium.dev/selenium/web/

harsha509 commented on examples/ruby/spec/drivers/options_spec.rb:
```diff
@@ -45,5 +45,50 @@
+      options.set_window_rect = true
+
+      driver = Selenium::WebDriver.for :firefox, options: options
+      driver.get('https://www.google.com')
```
can we use https://www.selenium.dev/selenium/web/ for all navigations etc.

harsha509 commented:
> Hello @diemol one question about the documentation, does it make sense that the timeouts are under the options page and not on its page?
> 
> If it does, I can also update the timeouts example on this PR
> 
> <img alt="Screenshot 2024-04-19 at 20 40 42" width="1192" src="https://private-user-images.githubusercontent.com/33221555/324081481-b9754131-2987-4cf2-b371-e45433e48c4f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5NzA2NzcsIm5iZiI6MTcxMzk3MDM3NywicGF0aCI6Ii8zMzIyMTU1NS8zMjQwODE0ODEtYjk3NTQxMzEtMjk4Ny00Y2YyLWIzNzEtZTQ1NDMzZTQ4YzRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI0VDE0NTI1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM2MzY4NDZlZjc1OTI5NTFlYTZlMjhmZjE5OWRkZWUzMjRkNDYzMjk0Zjg1NTdiNTlmNTZiM2E0NzUyMzFiYWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Pdl_CzXM-JDfpwwT3UGo6x4S43RUarbfvMywPvo-NQ4">

Hi @aguspe,

The timeout mentioned refers to the W3C capability timeout in W3C WebDriver, see https://www.w3.org/TR/webdriver2/#capabilities

Also, It covers features like `ScriptTimeout`, `PageLoadTimeout`, and `ImplicitWaitTimeout`, as per the WebDriver documentation. see https://www.w3.org/TR/webdriver2/#timeouts

So, I believe it's correctly included in the driver options section.

Thanks,
Sri

aguspe committed:
Merge branch 'trunk' into expand_options_example_page

aguspe committed:
Update timeouts examples

aguspe committed:
Merge remote-tracking branch 'origin/expand_options_example_page' into expand_options_example_page

aguspe commented:
> > Hello @diemol one question about the documentation, does it make sense that the timeouts are under the options page and not on its page?
> > If it does, I can also update the timeouts example on this PR
> > <img alt="Screenshot 2024-04-19 at 20 40 42" width="1192" src="https://private-user-images.githubusercontent.com/33221555/324081481-b9754131-2987-4cf2-b371-e45433e48c4f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5NzA2NzcsIm5iZiI6MTcxMzk3MDM3NywicGF0aCI6Ii8zMzIyMTU1NS8zMjQwODE0ODEtYjk3NTQxMzEtMjk4Ny00Y2YyLWIzNzEtZTQ1NDMzZTQ4YzRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI0VDE0NTI1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM2MzY4NDZlZjc1OTI5NTFlYTZlMjhmZjE5OWRkZWUzMjRkNDYzMjk0Zjg1NTdiNTlmNTZiM2E0NzUyMzFiYWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Pdl_CzXM-JDfpwwT3UGo6x4S43RUarbfvMywPvo-NQ4">
> 
> Hi @aguspe,
> 
> The timeout mentioned refers to the W3C capability timeout in W3C WebDriver, see https://www.w3.org/TR/webdriver2/#capabilities
> 
> Also, It covers features like `ScriptTimeout`, `PageLoadTimeout`, and `ImplicitWaitTimeout`, as per the WebDriver documentation. see https://www.w3.org/TR/webdriver2/#timeouts
> 
> So, I believe it's correctly included in the driver options section.
> 
> Thanks, Sri

Thank you so much for the help I have updated this PR to cover the timeout examples :)

aguspe committed:
Update all navigations

aguspe committed:
Merge branch 'trunk' into expand_options_example_page

aguspe committed:
Update options after updating the tests

aguspe committed:
Merge remote-tracking branch 'origin/expand_options_example_page' into expand_options_example_page

aguspe committed:
Merge branch 'trunk' into expand_options_example_page

aguspe committed:
Merge branch 'trunk' into expand_options_example_page

aguspe commented:
Hi @harsha509 before this gets merged I need to update some lines because I added a new let on the Rspec test so I'm missing some updates

harsha509 commented:
HI @aguspe ,

can you do it in a new PR if possible ? let me know

Thanks,
Sri

aguspe committed:
Update options after updating the tests

aguspe committed:
Fix documentation bug

aguspe committed:
Merge remote-tracking branch 'origin/expand_options_example_page' into expand_options_example_page

aguspe commented:
> HI @aguspe ,
> 
> can you do it in a new PR if possible ? let me know
> 
> Thanks, Sri

I just updated it in a commit here and I fixed a documentation bug, I hope that's okey

aguspe committed:
Merge branch 'trunk' into expand_options_example_page

aguspe committed:
Merge branch 'trunk' into expand_options_example_page

aguspe committed:
Merge branch 'trunk' into expand_options_example_page

aguspe commented:
Hello @harsha509 would it be possible to merge this PR? I will create a new PR to expand the selenium docs and correct an issue with the upgrade selenium ruby example but I keep merging trunk into this PR because it gets out of sync :)

Thank you so much

aguspe commented:
This PR needs to be merged before #1695

harsha509 commented:
Thank you @aguspe !
